server/src/model.ts

```diff
@@ -6093,3 +6409,572 @@ export async function getPreferredFolderView(loggedInUserId: Uint8Array) {
+}) {
+  await mustBeAdmin(loggedInUserId);
+  const libraryId = await getLibraryAccountId();
+  const { sourceId } = await prisma.libraryActivityInfos.findUniqueOrThrow({
+    where: {
+      activityId: activityId,
+    },
+    select: {
+      sourceId: true,
+    },
+  });
+
+  await prisma.content.update({
+    where: {
+      id: activityId,
+      isPublic: true,
+      isFolder: false,
+      isDeleted: false,
+      ownerId: libraryId,
+      libraryActivity: {
+        status: LibraryStatus.PUBLISHED,
+      },
+    },
+    data: {
+      isPublic: false,
+      libraryActivity: {
+        update: {
+          // TODO: should we use the pending review status here or another status?
+          status: LibraryStatus.PENDING_REVIEW,
+        },
+      },
+      libraryActivityEvents: {
+        create: {
+          sourceId,
+          eventType: LibraryEventType.UNPUBLISH,
+          dateTime: new Date(),
+          userId: loggedInUserId,
+        },
+      },
+    },
+  });
+}
+
+// TODO: Notify owner that their activity needs revision before it will be accepted into the library
+export async function markLibraryRequestNeedsRevision({
+  sourceId,
+  comments,
+  userId,
+}: {
+  sourceId: Uint8Array;
+  comments: string;
+  userId: Uint8Array;
+}) {
+  await mustBeAdmin(userId);
+
+  await prisma.content.update({
+    where: {
+      id: sourceId,
+      isPublic: true,
+      isFolder: false,
+      isDeleted: false,
+      librarySource: {
+        status: LibraryStatus.PENDING_REVIEW,
+        ownerRequested: true,
+      },
+    },
+    data: {
+      librarySource: {
+        update: {
+          status: LibraryStatus.NEEDS_REVISION,
+          comments,
+        },
+      },
+      librarySourceEvents: {
+        create: {
+          eventType: LibraryEventType.MARK_NEEDS_REVISION,
+          dateTime: new Date(),
+          userId,
+          comments,
+        },
+      },
+    },
+  });
+}
+
+// TODO: Notify owner that the comments have changed
+export async function modifyCommentsOfLibraryRequest({
+  sourceId,
+  comments,
+  userId,
+}: {
+  sourceId: Uint8Array;
+  comments: string;
+  userId: Uint8Array;
+}) {
+  await mustBeAdmin(userId);
+
+  const { activityId } = await prisma.libraryActivityInfos.findUniqueOrThrow({
+    where: {
+      sourceId,
+    },
+    select: {
+      activityId: true,
+    },
+  });
+
+  await prisma.content.update({
+    where: {
+      id: sourceId,
+      isPublic: true,
+      isFolder: false,
+      isDeleted: false,
+      NOT: {
+        librarySource: null,
+      },
+    },
+    data: {
+      librarySource: {
+        update: {
+          comments,
+        },
+      },
+      librarySourceEvents: {
+        create: {
+          activityId,
+          eventType: LibraryEventType.MODIFY_COMMENTS,
+          dateTime: new Date(),
+          comments,
+          userId,
+        },
+      },
+    },
+  });
+}
+
+export async function getCurationFolderContent({
```

Comment:
docstring

Reply:
Can we reuse `getMyFolderContent`? E.g., check for admin then call `getMyFolderContent` while lying who is `loggedInUserId`? (Maybe that one needs to be `ownerId`.)

Reply:
It looks like you reused `searchMyFolderContent`. Could reuse `getMyFolderContent` in similar way.